algorithms/src/crh/bowe_hopwood_pedersen.rs

```diff
@@ -106,30 +110,21 @@ impl<G: Group, S: PedersenSize> CRH for BoweHopwoodPedersenCRH<G, S> {
                 S::NUM_WINDOWS,
             ));
         }
-
-        // Pad the input if it is not the current length.
-        let mut input_bytes = input;
-        let mut padded_input_bytes = vec![];
-        if (input.len() * 8) < S::WINDOW_SIZE * S::NUM_WINDOWS {
-            padded_input_bytes.extend_from_slice(input_bytes);
-            padded_input_bytes.resize((S::WINDOW_SIZE * S::NUM_WINDOWS) / 8, 0u8);
-            input_bytes = padded_input_bytes.as_slice();
+        // we cant use these in array sizes since they are from a trait (and cant be refered to at const time)
+        assert!(S::WINDOW_SIZE <= 256);
+        assert!(S::NUM_WINDOWS <= 296);
+
+        // overzealous but stack allocation
+        let mut buffer = [0u8; 296 * 256 / 8 + BOWE_HOPWOOD_CHUNK_SIZE + 1];
```

Comment:
@Protryon what do you think about setting the 256 and 296 as global consts in this file, so it's clearly denoted at the top that these are magical constants and aren't buried into the code itself?

Reply:
Sounds reasonable